haystack/preview/dataclasses/answer.py

```diff
@@ -0,0 +1,26 @@
+from typing import Any, Dict, List, Optional
+from dataclasses import dataclass
+from haystack.preview.dataclasses.document import Document
+
+
+@dataclass(frozen=True)
+class Answer:
+    data: Any
+    question: str
+    metadata: Dict[str, Any]
+
+
+@dataclass(frozen=True)
+class ExtractiveAnswer(Answer):
+    data: Optional[str]
+    document: Document
+    probability: float
+    start: Optional[int] = None
+    end: Optional[int] = None
+
+
+@dataclass(frozen=True)
+class GenerativeAnswer(Answer):
+    data: str
+    documents: List[Document]
+    probability: float
```

Comment:
Just a curiosity: do you think there's a univocal meaning for this field in generative answer objects? Do LLMs always also output probability scores?

If it could be used, but its meaning change from LLM to LLM and they can't be made comparable, I'd rather use `metadata` to store this value. If instead we're fairly confident we can keep it.

Reply:
Basically all classification models (where next token prediction of generative models also counts as classification) use softmax at the end. Therefore, they always output a pseudo-probability (all scores between 0 and 1 and adding up to 1) and I would actually consider them less ambiguous than in the extractive case.
However, I just had a look at the OpenAI API specs and there doesn't seem to be a way to request the probability. This is probably done because the probability would always be very low with a large number of tokens which would be confusing to people (because a lot of probabilities are multiplied with each other) so I would think I should probably remove this field.

Reply:
Ok, good to know! This is the exact case where I would make the components put this value in `metadata`, rather than having a dedicated field.